python/TestHarness/testers/Tester.py

```diff
@@ -71,21 +71,65 @@ def __init__(self, name, params):
 
     # Set the status message
     if self.specs['check_input']:
-      self.status_message = 'SYNTAX PASS'
+      self.success_message = 'SYNTAX PASS'
     else:
-      self.status_message = self.specs['status_message']
+      self.success_message = self.specs['success_message']
 
     # Set up common paramaters
     self.should_execute = self.specs['should_execute']
     self.check_input = self.specs['check_input']
+    self.success_bucket = 'PASS'
+
+    # Initialize the status bucket
+    self.status = {}
+
+    # Initialize the tester with a passing status
+    self.setStatus(self.success_message, self.success_bucket)
+
+  # Return text color based on bucket status
+  def getColor(self, status):
+    BUCKET_COLOR = { 'PASS'    : 'GREEN',
+                     'FAIL'    : 'RED',
+                     'DIFF'    : 'YELLOW',
+                     'PBS'     : 'CYAN',
+                     'PENDING' : 'YELLOW',
+                     'DELETED' : 'RED',
+                     'SKIP'    : 'RESET'}
+
+    if status in BUCKET_COLOR:
+      return BUCKET_COLOR[status]
+    return 'RESET'
 
   # Method to return the input file if applicable to this Tester
   def getInputFile(self):
     return None
 
+  # Method to return the successful message printed to stdout
+  def getSuccessMessage(self):
+    return self.success_message
+
+  # Method to return status text (exodiff, crash, skipped because x, y and z etc)
   def getStatusMessage(self):
     return self.status_message
 
+  # Method to return enumerated status (PASS, FAIL, SKIP, DIFF, SILENT, DELETED)
+  def getStatus(self):
+    if self.getStatusMessage() in self.status:
+      return self.status[self.getStatusMessage()]
+    return None
+
+  # Method to set the bucket status
+  def setStatus(self, reason, bucket):
+    self.status_message = reason
+    self.status[reason] = bucket
+    return self.getStatus()
+
+  # Method to check for successfull test
+  def didPass(self):
+    if self.getStatus() == self.success_bucket:
+      return True
+    return False
```

Comment:
Ok man... I know you said there is a better way. I looked, and can't figure out how:

```python
def didPass(self):
  return self.getStatus()  # or self.get<whatever you mentioned>
```
is going to work for every case scenario...